### ISSUE TITLE
DotNetCoreHostEnvironment is never used

### ISSUE THREAD

**Issue by cremor** [Topshelf/Topshelf#583]
In commit 77548a22509bc21e1fc8e8f8bac99735bf0765b8 @slaneyrw removed the usage of the class `DotNetCoreEnvironmentBuilder` from the class `HostConfiguratorImpl`. This is very strange since the same commit also heavily expanded the class `DotNetCoreHostEnvironment`, and that class is also not used any more now.

It's still possible to use the code by manually changing the configuration to use it as shown in https://github.com/Topshelf/Topshelf/issues/513#issuecomment-499339087. But I suspect that this is not the indended use case, right? No user would know about this unless he reads the source code or finds this exact issue.

**Comment by phatboyg** [comment]
I am closing all issues that haven't seen any activity. You're welcome to submit a PR to resolve the issue or provide more information.

**Comment by cremor** [comment]
@phatboyg What information is missing here?

Sure, I could provide a PR. It's a one line fix. But I had hoped that @slaneyrw would add something here since the change was done by him. Maybe it was on purpose and I'm missing something.

**Comment by slaneyrw** [comment]
Sorry, missed the notification.

The expanded DotNetCoreHostEnvironment was my first attempt to try and get parity between the Full Framework Windows and .NET Core, as all the methods return false.

When I got to the installer commands, I found no direct .NET Core ( at least at the time, back in 1.x or 2.x ) libraries that could do the service installer part.

So when I put in the explicit ServiceInstaller port ( in TS.ServiceInstaller ) library there was no need to continue to use the DotNetCoreHostEnvironment anymore.

The situation may have changed for .NEt Core 3.x,and/or .NET 5.x, but it was the case for earlier versions of .NET Core.

I propably should have either removed DotNetCoreHostEnvironment  or put it back to NotImplement.

Should TopShelf ever support systemd on non-windows platform, or use the generic Host engine, then maybe we need to revisit DotNetCoreHostEnvironment
